### PR TITLE
Pala-customer-survey

### PR DESCRIPTION
**Pull request by elbines**
Sakura Tanaka, Mikael Öling and Elba Cacan.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,32 +6,52 @@
...
-The command below is a combination of installing dependencies, opening up the project on VS Code and it will run a development server on your terminal.
 
-```bash
-npm i && code . && npm run dev
-```
 
 ### The Problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Due to varying commitments and differences in time zones, sceduling meetings was somewhat challenging. Despite this, a mutual understanding of our goal and the project requirements ensured we got the project done. A project we are happy with.
```
Comment (BeckieMorton) [inline README.md:27]: this is great to hear, that even though you found it hard with finding time to meet that you are happy with your project and were all on the same page :-)

File: README.md [hunk 2]
```diff
@@ -6,32 +6,52 @@
...
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Due to varying commitments and differences in time zones, sceduling meetings was somewhat challenging. Despite this, a mutual understanding of our goal and the project requirements ensured we got the project done. A project we are happy with.  
+
+Given the varied commitments each of us had this week, the amount of tasks differed between us. This wasn't a reflection of effort or dedication, but rather the reality of our individual scedules and time constraits. 
 
 ### View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
 
+https://pala-customer-survey.netlify.app/
```
Comment (BeckieMorton) [inline README.md:34]: LOVE the styling, it looks clean and the colour pallete is great! It was a great idea to put a link to Mikael's company as the end button

File: src/Components/Button/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import previousQuestionBtnIcon from "/assets/arrow_left.svg";
```
Comment (BeckieMorton) [inline src/Components/Button/Button.jsx:1]: I would love a comment in this .jsx file just stating what it does (I know it is pretty obvious given that it is called button though!

File: src/Components/MultiStepForm/MultiStepForm.jsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+            />
+          </div>
+        </form>
+      </div>
+      <Footer />
+    </div>
+  );
+};
+
+//***************************************************/
```
Comment (BeckieMorton) [inline src/Components/MultiStepForm/MultiStepForm.jsx:104]: not sure what all this code that is commented out is left here for?

File: src/Components/PrimaryReason/PrimaryReason.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import style from "./PrimaryReason.module.css";
+import { SecondaryHeading } from "../SecondaryHeading/SecondaryHeading";
+import { useState } from "react";
+import { ErrorMessage } from "../ErrorMessage/ErrorMessage";
+
+export const PrimaryReason = ({ updateFormData, value, error }) => {
```
Comment (BeckieMorton) [inline src/Components/PrimaryReason/PrimaryReason.jsx:6]: looking at how you pass props in your components is super helpful for my understanding, I find this difficult, thanks

File: src/Components/Suggestions/Suggestions.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import style from "./Suggestions.module.css";
+import { SecondaryHeading } from "../SecondaryHeading/SecondaryHeading";
+import { ErrorMessage } from "../ErrorMessage/ErrorMessage";
+
+export const Suggestions = ({ updateFormData, value, error }) => {
+  const thirdQuestion = (e) => updateFormData("thirdQuestion", e.target.value);
+
+  return (
+    <div>
+      <SecondaryHeading question="3. What suggestions do you have to improve your shopping experience?" />
```
Comment (BeckieMorton) [inline src/Components/Suggestions/Suggestions.jsx:10]: Your structure, with many modules doing small things seems really well done, something to work towards. I would not think to do such small functions for eg. SecondaryHeading...but it makes sense as this is repeated. good job

File: src/Components/MultiStepForm/MultiStepForm.jsx [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    console.log(currentStep);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+  };
+
+  // Erro handling this is when a user doesn't fill, then a next button doesn't work.
+  useEffect(() => {
+    if (currentStep === 1)
```
Comment (BeckieMorton) [inline src/Components/MultiStepForm/MultiStepForm.jsx:46]: nice error handling and user input validation

File: src/Components/MultiStepForm/MultiStepForm.jsx [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+      formData.thirdQuestion === ""
+        ? setError(() => ({ message: "Please write your suggestions 🤲", isFailed: true }))
+        : setError(() => ({ message: "", isFailed: false }));
+  }, [currentStep, formData]);
+
+  return (
+    <div className={style.wrapper}>
+      <div className={style.form_wrapper}>
+        <img src="/assets/logo.png" className={style.logo} />
+        {/*Mikael edited this to get hero to work*/}
```
Comment (BeckieMorton) [inline src/Components/MultiStepForm/MultiStepForm.jsx:64]: these types of comments, I think, are useful in a group project

**Review by BeckieMorton — COMMENTED** [review]
Your styling and design look great! simple colour pallette and a simple easy user interface. Your code is clean, and the structure of modules is excellent (your file structure and how you pass probs and use many modules was help for me to see and has taught me). You use self-explanitory variable and function names but I would love some more comments throughout your program, would just help a beginner to follow a bit easily. Well done, guys!

Oh one last thing! Perhaps a bit more information in your pull request in the heading (it didn't have your names) would have made it just that much easier to locate :-) Thanks!

File: README.md (hunk 1 above)
Comment (elbines) [reply README.md:27]: Thank you for the thorough  and helpful code review! Not just this comment, but all of them :)

File: src/Components/MultiStepForm/MultiStepForm.jsx [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+
+  // Erro handling this is when a user doesn't fill, then a next button doesn't work.
+  useEffect(() => {
+    if (currentStep === 1)
+      formData.primaryReason === ""
+        ? setError(() => ({ message: "Please choose one option ⭐️", isFailed: true }))
+        : setError(() => ({ message: "", isFailed: false }));
+    if (currentStep === 2)
+      formData.howOften === ""
+        ? setError(() => ({ message: "Please select one 🦮", isFailed: true }))
```
Comment (Soygirt) [inline src/Components/MultiStepForm/MultiStepForm.jsx:52]: I like the error message as well, love that you used emojis in them all!
Me personally would prefer if it wasn't showing from the start, or that it changed when I tried to skip choosing an answer 😄 Maybe I'm just slow, but I didn't notice that it was an error message at first, and thought it was a little strange that you didn't have any 😅

File: src/Components/PrimaryReason/PrimaryReason.jsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+            <label htmlFor="other">Other</label>
+          </div>
+          {isOther && (
+            <input
+              type="text"
+              id="other"
+              className={style.text_input}
+              name="primaryReason"
+              onChange={firstQuestion}
+              placeholder="Other..."
```
Comment (Soygirt) [inline src/Components/PrimaryReason/PrimaryReason.jsx:90]: Really neat that this text input pops up when you choose "other"!

File: src/Components/Summary/Summary.jsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+      <SecondaryHeading question="Survey Summary" />
+      <div className={style.inner_box}>
+        <h5>The primary reason:</h5>
+        <p>{data.primaryReason}</p>
+        <h5>How often you made an order:</h5>
+        <p>{data.howOften}</p>
+        <h5>Suggestions: </h5>
+        <p>{data.thirdQuestion}</p>
+        <SubmitButton text="Submit" onClickEvent={nextStep} />
+      </div>
```
Comment (Soygirt) [inline src/Components/Summary/Summary.jsx:17]: This is a clever way to sum up the answered questions and give the user an overview on his/her answers so they have the opportunity to go back and change if somthing is wrong.

File: src/Components/Thankyou/Thankyou.jsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <>
+      <div className={style.inner}>
+        <SecondaryHeading question="Thank you for sharing your feedback with us!" />
+        <p className={style.text}>Your €15 Discount Code: CXD48SS</p>
+        <SubmitButton
+          text="Exit Survey"
+          onClickEvent={handleClick}
+          url="https://palapets.com/discount/CXD48SS"
```
Comment (Soygirt) [inline src/Components/Thankyou/Thankyou.jsx:18]: I was surprised when I got to the palapets website. Didn't expect that! But I think this is a better better way to end the survey then for the user to just close the window or be left on the survey-page! I'd never thought about this solution! Great job!

**Review by Soygirt — COMMENTED** [review]
Wow! What a great project you've made! I really like the styling and that you used an existing company for the project. 
I would have liked some more comments to be able to understand all the functions a little better. But all functions are very well named so it's quite easy to follow anyways :)

Great job you guys! 👏

**Pull request closed without merging**